### DISCUSSION TITLE
Glob markdown files on github action only returns `docs` folder

### DISCUSSION THREAD

**Discussion by a11smiles** [community/community#25619] [Actions]
<p>I have multiple folders in my project containing markdown files. I want to run <code>cspell</code> (and other tools) against them. Unfortunately, only the markdown files in the <code>docs</code> subfolder are returned in the blob.</p>
<p>Here’s a snippet of an actions file:</p>
<pre><code class="lang-auto">      - name: 'Spellchecking Content'
        run: |
          npm install cspell -g
          pwd
          ls ./**/*.md
          npx cspell $(ls ./**/*.md) -c ${{ github.workspace }}/cspell.json
        working-directory: ${{ github.workspace }}

</code></pre>
<p>Again, the problem is that the <code>ls ./**/*.md</code> in GitHub actions <strong>only returns the markdown files in the <code>docs</code> folder</strong> and no markdown files in the other subfolders.</p>
<p><strong>Troubleshooting:</strong></p>
<ul>
<li>
<code>pwd</code> prints the correct, home folder of the project workspace</li>
<li>
<code>ls ./**/*.md</code> only lists the files in the <code>docs</code> subfolder</li>
<li>I’ve attempted to set the <code>working-directory</code> explicitly</li>
<li>The command works on my local machine, just not on GitHub</li>
</ul>

**Comment by a11smiles** [comment]
<p>So, interestingly enough, if I replace</p>
<pre><code class="lang-auto">npx cspell $(ls ./**/*.md) -c ${{ github.workspace }}/cspell.json
</code></pre>
<p>with</p>
<pre><code class="lang-auto">npx cspell $(find . -name '*.md') -c ${{ github.workspace }}/cspell.json
</code></pre>
<p>it works.</p>
<p>Is this a bug or documented limitation in GitHub actions?</p>

**Comment by ChristopherHX** [comment, marked as answer]
<p>It seems the bash in github actions has different defaults like your system,<br>
You might need to add <code>shopt -s globstar</code> to your script before <code>ls ./**/*.md</code> to enable the globstar.<br>
E.g.</p>
<pre><code class="lang-yaml">      - name: 'Spellchecking Content'
        run: |
          shopt -s globstar
          npm install cspell -g
          pwd
          ls ./**/*.md
          npx cspell $(ls ./**/*.md) -c ${{ github.workspace }}/cspell.json
        working-directory: ${{ github.workspace }}

</code></pre>
<p>See here for more information:</p><aside class="quote" data-post="4" data-topic="198474">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/path-patterns-not-working/198474/4">Path patterns not working</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    Each step gets a fresh shell, so changes to shell settings are lost after the step. You need to set the globstar option in the step where you want to use the ** glob. 
Try: 
      - run: |
          shopt -s globstar
          ls tests/**/*.py
  </blockquote>
</aside>

**Comment by a11smiles** [comment]
<p>that works. thanks.</p>
<p>I also found that the <code>find</code> command is a suitable alternative too.</p>
<p>thanks again!</p>
